hathorlib/nanocontracts/nanocontract.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import struct
+from typing import List, NamedTuple, Optional, Tuple
+
+from hathorlib.transaction import Transaction
+from hathorlib.base_transaction import TxInput, TxOutput, TxVersion
+from hathorlib.utils import int_to_bytes, unpack, unpack_len
+
+
+NC_VERSION = 1
+
+
+class NCCallInfo(NamedTuple):
+    """This tuple carries the pieces of information serialized inside transactions."""
+    version: int
+    id: bytes
+    method: str
+    args_bytes: bytes
+    pubkey: bytes
+    signature: bytes
+
+
+class NanoContract(Transaction):
```

Comment:
I miss unit tests for the serialization and deserialization methods.

Reply:
I will add them and fix the linter

Reply:
Done

Reply:
Thanks!